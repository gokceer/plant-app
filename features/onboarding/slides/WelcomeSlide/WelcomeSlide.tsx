import React from "react";
import { Dimensions, Image, StyleSheet, Text, View } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

const { width: screenWidth, height: screenHeight } = Dimensions.get("window");

const WelcomeSlide = () => {
  return (
    <SafeAreaView>
      <View style={styles.header}>
        <Text style={styles.title}>
          Welcome to <Text style={styles.titleBold}>PlantApp</Text>
        </Text>
        <Text style={styles.subtitle}>
          {"Identify more than 3000+ plants with\n88% accuracy."}
        </Text>
      </View>
      <Image
        source={require("../../../../assets/images/welcomeSlidePlant.png")}
        style={styles.image}
        accessibilityLabel="Plant"
      />
    </SafeAreaView>
  );
};

export default WelcomeSlide;

const styles = StyleSheet.create({
  header: {
    marginTop: 12,
    marginBottom: 24,
    marginLeft: 24,
  },
  title: {
    fontSize: 28,
    color: "#000",
  },
  titleBold: {
    fontFamily: "RubikBold",
  },
  subtitle: {
    fontFamily: "RubikRegular",
    fontSize: 16,
    marginTop: 8,
    color: "#13231BB2",
  },
  image: {
    width: screenWidth,
    height: screenHeight * 0.65,
    resizeMode: "contain",
    alignSelf: "center",
  },
  buttonWrapper: {
    marginHorizontal: 24,
  },
  termsText: {
    fontSize: 11,
    color: "#597165B2",
    textAlign: "center",
    lineHeight: 14,
    fontFamily: "RubikRegular",
    marginTop: 16,
  },
  linkText: {
    textDecorationLine: "underline",
  },
});
