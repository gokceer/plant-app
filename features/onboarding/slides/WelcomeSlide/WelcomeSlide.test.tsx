import { render } from "@testing-library/react-native";
import React from "react";
import WelcomeSlide from "./WelcomeSlide";

describe("WelcomeSlide", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<WelcomeSlide />);
    expect(toJSON()).toMatchSnapshot();
  });
});
