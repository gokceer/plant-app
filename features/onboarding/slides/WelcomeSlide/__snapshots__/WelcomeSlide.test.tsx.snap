// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSlide renders correctly and matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "marginBottom": 24,
        "marginLeft": 24,
        "marginTop": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 28,
        }
      }
    >
      Welcome to 
      <Text
        style={
          {
            "fontFamily": "RubikBold",
          }
        }
      >
        PlantApp
      </Text>
    </Text>
    <Text
      style={
        {
          "color": "#13231BB2",
          "fontFamily": "RubikRegular",
          "fontSize": 16,
          "marginTop": 8,
        }
      }
    >
      Identify more than 3000+ plants with
88% accuracy.
    </Text>
  </View>
  <Image
    accessibilityLabel="Plant"
    source={
      {
        "testUri": "../../../assets/images/welcomeSlidePlant.png",
      }
    }
    style={
      {
        "alignSelf": "center",
        "height": 867.1,
        "resizeMode": "contain",
        "width": 750,
      }
    }
  />
</RNCSafeAreaView>
`;
